ffs/minerselector/sr2/sr2.go

```diff
@@ -41,8 +39,8 @@ type bucket struct {
 }
 
 // New returns a new SR2 miner selector.
-func New(url string, ai *askindex.Runner, cb lotus.ClientBuilder) (*MinerSelector, error) {
-	ms := &MinerSelector{url: url, ai: ai, cb: cb}
+func New(url string, cb lotus.ClientBuilder) (*MinerSelector, error) {
```

Comment:
Ugly diff, but basically:
- Always query ask miners in the list, since they're changing prices fast and the cache might give stalled values. Also, the ask index might be disabled temporarily so can also have stalled values by other reason.
- Take into account `MaxPrice` if set. If a miner doesn't satisfy that constraint, is skipped and it looks for a next one.